### PR TITLE
implement ticket 7-post-comments

### PR DESCRIPTION
**Pull request by akashdeep931**

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -178,3 +178,42 @@ describe("GET /api/reviews/:review_id/comments", () => {
...
+      .send({ username: "bainesface", body: "Nice picture!" })
+      .expect(201)
+      .then(({ body }) => {
+        expect(body).toHaveProperty("comment_id", expect.any(Number));
+        expect(body).toHaveProperty("body", expect.any(String));
+        expect(body).toHaveProperty("review_id", expect.any(Number));
+        expect(body).toHaveProperty("author", expect.any(String));
+        expect(body).toHaveProperty("votes", expect.any(Number));
+        expect(body).toHaveProperty("created_at", expect.any(String));
+      });
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/integration.test.js:195]: Good test! To make this test more robust, can we expect a specific value for the `body`, `review_id`, and `author` properties based on the request we are making ?

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -178,3 +178,42 @@ describe("GET /api/reviews/:review_id/comments", () => {
...
+      .send({})
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+
+        expect(msg).toBe("Bad Request!");
+      });
+  });
+  it("400: should return an error when given an object with wrong values", () => {
+    return request(app)
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/integration.test.js:209]: for this error, psql is unable to find the `username` in the `users` table, which would make the code `404 - Not Found`

File: __tests__/integration.test.js [hunk 3]
```diff
@@ -178,3 +178,42 @@ describe("GET /api/reviews/:review_id/comments", () => {
...
+      .post("/api/reviews/1/comments")
+      .send({ username: "kAKASHi", body: "This should response with an error" })
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+
+        expect(msg).toBe("Bad Request!");
+      });
+  });
+});
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/integration.test.js:219]: Really good testing! There are a few other scenarios that we must also include tests for:
- What if we are making a request with additional properties in the post body ? Will your model ignore those unnecessary properties and still be able to make a successful post using only the relevant values
- What if the review ID is not a number 
- What if the review ID cannot be found in the database 

In the seed file you may notice that the `comments` table has `review_id` as a foreign key, so psql may be able to throw some of those errors for you

File: __tests__/integration.test.js (hunk 1 above)
Comment (akashdeep931) [reply __tests__/integration.test.js:195]: Mmm yes, we can expect the defined keys to have the values of the object sent by the client. Is there any variable to aim the new item or I just write it manually like expect(body.body).toBe("whatever we have written in the body property")?

File: controllers/post.controllers.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const { insertComment } = require("../models/post.models.js");
+
+exports.postComment = (req, res, next) => {
+  const { review_id } = req.params;
+  const body = req.body;
+
+  insertComment(review_id, body)
+    .then((data) => {
+      res.status(201).send(data);
```
Comment (Pipe-Smoking-Rabbit) [inline controllers/post.controllers.js:9]: when we send a response from our server, we want that data to be in an `object` stored on an appropriately named `key`. Currently you are just sending `data`, would it be possible to send the data on a key, i.e. - `.send({ comment: <Your Data Here> });` ?

Comment (Pipe-Smoking-Rabbit) [reply controllers/post.controllers.js:9]: _**Note:**_ _this will affect how you access the data in the testing _

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Hey, very nice code it's coming along really well! 
I have left some feedback for suggested changes, most importantly of which is a suggestion for additional tests that must be added to the endpoint. Once you have added those tests, send in another nchelp pr and we can take a look! 👍

File: __tests__/integration.test.js (hunk 1 above)
Comment (Pipe-Smoking-Rabbit) [reply __tests__/integration.test.js:195]: I would imagine you could do that!

File: __tests__/integration.test.js [hunk 5]
```diff
@@ -178,3 +178,95 @@ describe("GET /api/reviews/:review_id/comments", () => {
...
+
+        expect(comment).toHaveProperty("comment_id", expect.any(Number));
+        expect(comment).toHaveProperty("body", expect.any(String));
+        expect(comment).toHaveProperty("review_id", expect.any(Number));
+        expect(comment).toHaveProperty("author", expect.any(String));
+        expect(comment).toHaveProperty("votes", expect.any(Number));
+        expect(comment).toHaveProperty("created_at", expect.any(String));
+
+        expect(comment.review_id).toBe(4);
+        expect(comment.author).toBe("bainesface");
```
Comment (callumu21) [inline __tests__/integration.test.js:199]: You can remove these properties above (lines 192-194) as these more specific tests cover them nicely

File: controllers/post.controllers.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  checkReviewExists,
+} = require("../models/post.models.js");
+
+exports.postComment = (req, res, next) => {
+  const { review_id } = req.params;
+  const body = req.body;
+
+  Promise.all([
+    checkReviewExists(review_id),
+    checkUsernameExists(body),
```
Comment (callumu21) [inline controllers/post.controllers.js:13]: As author is a foreign key in the comments table, this isn't strictly needed. PSQL will throw an error for us and we can use that and save a function call here.

**Review by callumu21 — APPROVED** [review]
Really nice! I've left a few comments that could help with refactoring, but merge away when you're ready 😄

**Pull request merged**